src/Model/MenuItem.php

```diff
@@ -41,7 +39,7 @@ class MenuItem extends Model {
 	 * @throws \Exception
 	 */
 	public function __construct( \WP_Post $post ) {
-		$this->post = $post;
+		$this->post = wp_setup_nav_menu_item( $post );
```

Comment:
Might be better to have this happen in `DataSource` 🤔

Reply:
🤔 Let me think about that a bit.

Reply:
Ya, I don't think so. DataSource now is essentially concerned with the Deferred resolution of nodes. The model is more concerned with actually determining what data should be returned to represent the node. 

Like here's the entire `DataSource::resolve_menu_item()` currently:

```
public static function resolve_menu_item( $id, AppContext $context ) {
		if ( empty( $id ) || ! absint( $id ) ) {
			return null;
		}
		$menu_item_id = absint( $id );
		$context->MenuItemLoader->buffer( [ $menu_item_id ] );

		return new Deferred( function () use ( $menu_item_id, $context ) {
			return $context->MenuItemLoader->load( $menu_item_id );
		} );
	}
```

It basically just asks the Loader to load the item.

The Loader then gets the object and passes it through the Model Layer, so it's the model layers job now to prepare the data for response (or return null)

Reply:
If there's a strong argument the other way, I'll consider, but I think the DataSource methods have a bit more of a clearer role now